### ISSUE TITLE
Adjusting UI scaling / scale factor 

### ISSUE THREAD

**Issue by x3ro** [hakolao/egui_winit_vulkano#23]
Hey there. First of all, thanks for this glue package, it's made playing around with `egui` much easier :)

I noticed that, running the `wholesome` example on macOS, that pixel density / HiDPI / whatever you want to call it was not taken into account. In other words, logical pixels and physical pixels were treated as the same thing, resulting in very small rendering. Note the disparity between the size of the window title and the content of the window:
 
<img width="612" alt="Screenshot 2022-08-26 at 19 05 19" src="https://user-images.githubusercontent.com/243719/186956381-bc8521fe-24c9-479c-bbb4-77b90121ab16.png">

I did find `scale_factor_override` in `WindowDescriptor`, but adjusting that only resulted in the size of the window changing. That is, if I tried to create a window 500x500px and set a `scale_factor_override` of `2`, then the window would be 1000x1000px, but the content was still the same size.

Investigating a bit further, I found that `egui_winit` handles this sort of scaling via the `set_pixels_per_point` method on `egui_winit::State`. This state is created in `integration::Gui::new`, and adding the following before returning from the constructor results in the (imo) "correct" content size:

```rust
egui_winit.set_pixels_per_point(surface.window().scale_factor() as f32);
```

<img width="612" alt="Screenshot 2022-08-26 at 19 08 37" src="https://user-images.githubusercontent.com/243719/186956852-065de01e-cb9d-4c22-894a-441a143b29b9.png">

So, long story short, this results in a bunch of questions from my side:

* Do you think it's desirable to add this functionality to the `egui_winit_vulkano` crate?
    * If so, I think it'd be handy to both detect the default as shown above, but also allow the user to chose the `pixels_per_point` manually, if desired.
* Could you elaborate what the  the current idea behind `scale_factor_override` is?

**Comment by hakolao** [comment]
Yea, this should not occur. Let me get back to you when I get a moment :) tonight or over the weekend

**Comment by hakolao** [comment]
Looks like I broke it when updating to new `egui_winit` whose api changed. I'll make a fix! @x3ro

**Comment by hakolao** [comment]
`scale_factor_override` is used in the window builder in `vulkano-utils`. I think it just tells winit what the scale factor should be (if it's set), but if we don't tell `egui_winit` what the scale factor is, then egui will look wrong.

`egui_winit` is already publicly exposed from `Gui` thus you can set the scale factor for egui later if you wish.

**Comment by hakolao** [comment]
Thanks for reporting and using this lib!

**Comment by x3ro** [comment]
Nice, thanks for the quick fix! :)
